docs(wasi): 07-27-23 notes (#1333)
WebAssembly/meetings: wasi/2023/WASI-07-27.md

In order to define the contents of Preview 2, it must be voted for inclusion by the WASI Subgroup.

Proposals may be added to WASI Preview 2 at any time until Preview 3.
Any questions?

**Conrad Watt**: It has to be virtualizable using a component. 

**Dan Gohman**: We require that all APIs are defined in WIT. There is a possibility that APIs could be awkward to virtualize, but by requiring things be expressed in WIT then we know that everything can be virtualized. 

**Dan Gohman**: I looked around and I found this unofficial rule that when something is posted we wait a week to vote on these things. We are not going to vote now, but we will vote on this process change next week. 2 weeks from now.

**Pat Hickey**: I put this on the agenda.

**Dan Gohman**: Should we document this 1 week wait before an agenda?

**Conrad Watts**: Up to the subgroup on rules. CG follows one day beforehand for a poll. 

**Deepti**: Just 24 hours before in CG.

 **Pat Hickey**: OK then we can follow this in the future but not change for this vote.

> **Dan Gohman**: We intend for this process to not change.

Reviewer: ```suggestion
```

I think the context for this last remark isn't recorded here, so I think it's best to remove it.

**Conrad Watt**: Your point about WIT IDL can be subtly be virtualized. The question of a side-effects of the virtual machine state.

> **Dan Gohman**: Modules import the linear memory. In wasi-threads, all of the threads have their own thread that imports the linear memory. That is a construct that is not possible to represent. 

Reviewer: ```suggestion
**Dan Gohman**: Today, most modules define and export a linear memory. In wasi-threads, all of the threads have their own module that imports the linear memory. That is a construct that is not possible to represent in WIT.
```

**Luke Wagner**: In theory you could have a u32 that the host could modify. That’s not readily virtualizable so we wouldn’t want this API. 

> **Pat Hickey**: Modules are a core webassmebly unit. There is no concept of exporting a memory from a component. WIT is the interface definition, then you have to implement it in terms of a component. A WIT IDL doesn’t express anything in terms of a core module. The only thing that exists in a module is the core ABI (part of canonical ABI).

Reviewer 2: ```suggestion
**Pat Hickey**: Modules are a core webassmebly unit. There is no concept of exporting a memory from a component. WIT is the interface definition, then you have to implement it in terms of a component. A WIT IDL doesn’t express anything in terms of a core module. The only thing that exists in a module is the canonical ABI.
```
